### ISSUE TITLE
clang-format 16 on JSON with leading blank line results in corrupted content.

### ISSUE THREAD

**Issue by JohnC32** [llvm/llvm-project#62228]
Given this JSON where the **first** line is blank:

```json

{
    "a": "foo",
    "b": []
}
    
```

using `_clang-format`:

```yaml
---
BasedOnStyle: Google
IndentWidth: 4
ColumnLimit: 0
...
```

clang-format 16 yields the following which is invalid JSON with an "x ="

    x = 
    {
        "a": "foo",
        "b": []
    }

A similar issue ("x =") was reported and fixed in https://github.com/llvm/llvm-project/issues/52705.

**Comment by JohnC32** [comment]
Just to be clear on the test.json to use, you need a starting blank line, which I numbered here:

    1 
    2 {
    3     "a": "foo",
    4     "b": []
    5 }

**Comment by owenca** [comment]
https://reviews.llvm.org/D148777
